doc/examples/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,303 @@
+        RCLCPP_ERROR(LOGGER, "Failed to find planning scene '%s'", scene_name.c_str());
+        return false;
+      }
+    }
+    catch (std::exception& ex)
+    {
+      RCLCPP_ERROR(LOGGER, "Error loading planning scene: %s", ex.what());
+      return false;
+    }
+
+    // Add object to planning scene
+    {  // Lock PlanningScene
+      planning_scene_monitor::LockedPlanningSceneRW scene(moveit_cpp_->getPlanningSceneMonitor());
+      scene->processPlanningSceneWorldMsg(scene_msg.world);
+    }  // Unlock PlanningScene
+
+    RCLCPP_INFO(LOGGER, "Loaded planning scene successfully");
+
+    // Get planning scene query
+    moveit_warehouse::MotionPlanRequestWithMetadata planning_query;
+    std::string query_name = scene_name + "_query";
+    try
+    {
+      planning_scene_storage->getPlanningQuery(planning_query, scene_name, query_name);
+    }
+    catch (std::exception& ex)
+    {
+      RCLCPP_ERROR(LOGGER, "Error loading motion planning query '%s': %s", query_name.c_str(), ex.what());
+    }
+
+    planning_query_request_ = static_cast<moveit_msgs::msg::MotionPlanRequest>(*planning_query);
+    return true;
+  }
+
+  /// \brief Set joint goal state for next planning attempt
+  /// \param [in] panda_jointN Goal value for the Nth joint [rad]
+  void setJointGoal(double const panda_joint1, double const panda_joint2, double const panda_joint3,
+                    double const panda_joint4, double const panda_joint5, double const panda_joint6,
+                    double const panda_joint7)
+  {
+    auto robot_goal_state = planning_component_->getStartState();
+    robot_goal_state->setJointPositions("panda_joint1", &panda_joint1);
+    robot_goal_state->setJointPositions("panda_joint2", &panda_joint2);
+    robot_goal_state->setJointPositions("panda_joint4", &panda_joint4);
+    robot_goal_state->setJointPositions("panda_joint5", &panda_joint5);
+    robot_goal_state->setJointPositions("panda_joint6", &panda_joint6);
+    robot_goal_state->setJointPositions("panda_joint7", &panda_joint7);
+    robot_goal_state->setJointPositions("panda_joint3", &panda_joint3);
+
+    // Set goal state
+    planning_component_->setGoal(*robot_goal_state);
+  }
+
+  /// \brief Set goal state for next planning attempt based on query loaded from the database
+  void setQueryGoal()
+  {
+    // Set goal state
+    if (!planning_query_request_.goal_constraints.empty())
+    {
+      planning_component_->setGoal(planning_query_request_.goal_constraints);
+    }
+    else
+      RCLCPP_ERROR(LOGGER, "Planning query request does not contain any goal constraints");
+  }
+
+  /// \brief Request a motion plan based on the assumption that a goal is set and print debug information.
+  void planAndPrint()
+  {
+    // Set start state as current state
+    planning_component_->setStartStateToCurrentState();
+
+    moveit_cpp::PlanningComponent::MultiPipelinePlanRequestParameters multi_pipeline_plan_request{
+      node_, { "ompl_rrtc", "pilz_lin", "chomp" }
+    };
```

Comment:
Might want to add a comment here saying that you could select a subset of the pipelines defined in the YAML file.

Also, is there a default behavior that will use them all?

Reply:
No, there is not. It is always necessary to name all pipelines to be used here. Might be a nice API feature though!